tests/pure_pursuit_tests.cpp

```diff
@@ -179,4 +170,34 @@ TEST_CASE( "Test get distance to point", "[distance_point]" )
         auto dist2 = test_class.get_distance_to_point_test( { 3, 10 } );
         CHECK( dist2 != -1 );
     }
+}
+
+TEST_CASE( "Test get lookahead point", "[lookahead_point]" )
+{
+    SECTION( "Simple Test" )
+    {
+        Path p = { { 0, 0, 0 }, { 10, 0, 10 } };
+        PurePursuitTest test_class( p, 3 );
+        auto point = test_class.get_lookahead_point_test( { 5, 5, 5 } );
+        CHECK( point == Point3D( 8, 0, 8 ) );
+
+        point = test_class.get_lookahead_point_test( { 3, 3, 3 } );
+        CHECK( point == Point3D( 6, 0, 6 ) );
+    }
```

Comment:
You should add a test case where the robot is on a path segment close to the end, and the lookahead point is on the next path segment.